### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by vitorpavani**

### CODE REVIEW

File: views/auth/login.hbs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<form id="form" action="/auth/login" method="POST">
+
+  <label for="username">Username</label><br />
```
Comment (CarlosZiegler) [inline views/auth/login.hbs:3]: Add attribute required

File: views/auth/login.hbs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  <input type="text" name="username" />
+
+  <label for="password">Password</label><br />
```
Comment (CarlosZiegler) [inline views/auth/login.hbs:6]: Here too with required. https://www.w3schools.com/tags/att_input_required.asp

File: routes/auth.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  passport.authenticate('local', {
+    successRedirect: '/',
+    failureRedirect: '/auth/login',
+    failureFlash: true,
+    passReqToCallback: true
+  })
+)
+
+// github authentication routing
+router.get('/auth/github', passport.authenticate('github'));
```
Comment (CarlosZiegler) [inline routes/auth.js:49]: Add annotations  from Swagger for Documentation Routes

File: routes/auth.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+router.get(
+  '/auth/github/callback',
+  passport.authenticate('github', {
+    successRedirect: '/',
+    failureRedirect: '/auth/login',
+  })
+);
+
+
```
Comment (CarlosZiegler) [inline routes/auth.js:59]: Add annotations  from Swagger for Documentation Routes

File: routes/auth.js [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+          res.redirect('/');
+        })
+        .catch(err => {
+          next(err);
+        });
+    }
+  });
+});
+
+
```
Comment (CarlosZiegler) [inline routes/auth.js:100]: Add annotations  from Swagger for Documentation Routes

File: routes/auth.js [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+router.get('/auth/login', (req, res) => {
+  res.render('auth/login');
+});
+
```
Comment (CarlosZiegler) [inline routes/auth.js:104]: Add annotations  from Swagger for Documentation Routes

File: views/auth/signup.hbs [hunk 7]
```diff
@@ -0,0 +1,16 @@
+<form id="form" action="/auth/signup" method="POST">
+
+  <label for="username">Username</label><br />
```
Comment (CarlosZiegler) [inline views/auth/signup.hbs:3]: Delete this </br> because you can add blank space with CSS space margin or padding

File: routes/auth.js [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+    res.render('auth/signup', {
+      message: 'Your password must be 8 characters minimun.'
+    });
+    return;
+  }
+  if (username === '') {
+    res.render('auth/signup', { message: 'Your username cannot be empty' });
+    return;
+  }
+
```
Comment (CarlosZiegler) [inline routes/auth.js:73]: Db queries inside an Try/Catch

File: routes/auth.js [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+  User.findOne({ username: username }).then(found => {
+    if (found !== null) {
+      res.render('auth/signup', { message: 'Wrong credentials' });
+    } else {
+      // we can create a user with the username and password pair
+      const salt = bcrypt.genSaltSync();
+      const hash = bcrypt.hashSync(password, salt);
+
```
Comment (CarlosZiegler) [inline routes/auth.js:81]: Db queries inside an Try/Catch

File: views/auth/login.hbs (hunk 1 above)
Comment (vitorpavani) [reply views/auth/login.hbs:3]: Done

File: views/auth/login.hbs (hunk 2 above)
Comment (vitorpavani) [reply views/auth/login.hbs:6]: Done

File: routes/auth.js (hunk 3 above)
Comment (vitorpavani) [reply routes/auth.js:49]: Done

File: routes/auth.js (hunk 4 above)
Comment (vitorpavani) [reply routes/auth.js:59]: Done

File: routes/auth.js (hunk 5 above)
Comment (vitorpavani) [reply routes/auth.js:100]: Done

File: routes/auth.js (hunk 6 above)
Comment (vitorpavani) [reply routes/auth.js:104]: Done

File: views/auth/signup.hbs (hunk 7 above)
Comment (vitorpavani) [reply views/auth/signup.hbs:3]: temporary form for testing only.

**Pull request merged**